opnode/cmd/main.go

```diff
@@ -1,31 +1,223 @@
 package main
 
 import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
 	"github.com/ethereum-optimism/optimistic-specs/opnode/node"
-	"github.com/protolambda/ask"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/urfave/cli"
+)
+
+var (
+	Version = "0.0.0"
+	// GitCommit   = ""
+	// GitDate     = ""
+	VersionMeta = "dev"
 )
 
-type MainCmd struct {
+// VersionWithMeta holds the textual version string including the metadata.
+var VersionWithMeta = func() string {
+	v := Version
+	if VersionMeta != "" {
+		v += "-" + VersionMeta
+	}
+	return v
+}()
+
+func main() {
+	// Set up logger with a default INFO level in case we fail to parse flags,
+	// otherwise the final critical log won't show what the parsing error was.
+	log.Root().SetHandler(
+		log.LvlFilterHandler(
+			log.LvlInfo,
+			log.StreamHandler(os.Stdout, log.TerminalFormat(true)),
+		),
+	)
+
+	app := cli.NewApp()
+	app.Flags = Flags
+	app.Version = VersionWithMeta
+	app.Name = "opnode"
+	app.Usage = "Optimism Rollup Node"
+	app.Description = "The deposit only rollup node drives the L2 execution engine based on L1 deposits."
+
+	app.Action = RollupNodeMain
+	err := app.Run(os.Args)
+	if err != nil {
+		log.Crit("Application failed", "message", err)
+	}
+}
+
+func RollupNodeMain(ctx *cli.Context) error {
+	log.Info("Initializing Rollup Node")
+	cfg, err := NewConfig(ctx)
+	if err != nil {
+		log.Error("Unable to create the rollup node config", "error", err)
+		return err
+	}
+	logCfg, err := NewLogConfig(ctx)
+	if err != nil {
+		log.Error("Unable to create the log config", "error", err)
+		return err
+	}
+
+	n, err := node.New(context.Background(), cfg, logCfg.NewLogger())
+	if err != nil {
+		log.Error("Unable to create the rollup node", "error", err)
+		return err
+	}
+	log.Info("Starting rollup node")
+
+	if err := n.Start(); err != nil {
+		log.Error("Unable to start L2 Output Submitter", "error", err)
+		return err
+	}
+	defer n.Stop()
+
+	log.Info("Rollup node started")
+
+	interruptChannel := make(chan os.Signal, 1)
+	signal.Notify(interruptChannel, []os.Signal{
+		os.Interrupt,
+		os.Kill,
+		syscall.SIGTERM,
+		syscall.SIGQUIT,
+	}...)
+	<-interruptChannel
+
+	return nil
+
+}
+
+// has0xPrefix validates str begins with '0x' or '0X'.
+// Copied from geth
+func has0xPrefix(str string) bool {
+	return len(str) >= 2 && str[0] == '0' && (str[1] == 'x' || str[1] == 'X')
 }
 
-func (c *MainCmd) Help() string {
-	return "Run Optimism rollup node."
+// HexToHash is copied from Geth, but does not supress the error
+func HexToHash(s string) (common.Hash, error) {
```

Comment:
```
	var x common.Hash
	err := x.UnmarshalText([]byte(stringhere))
```
May be a shorter alternative

Reply:
Yep, that's the what I want. The rest of the functions just seem to suppress errors.